### ISSUE TITLE
"No Cycle" SQL syntax is not cross-DB compatible

### ISSUE THREAD

**Issue by JayWelborn** [liquibase/liquibase#1990]
## Environment
**Liquibase Version**:
4.1.0
Also latest master

**Liquibase Integration & Version**:
gradle, unknown version

**Liquibase Extension(s) & Version**: 

**Database Vendor & Version**:
PostgreSQL: 12.X (occurs on all supported versions)

**Operating System Type & Version**:
MacOS 11.4 (20F71), but will occur on all supported OSs

## Description
A clear and concise description of the issue being addressed.
The SQL generated by the `alterSequence` change type is not correct syntax for all DBs. The alterSequence change type currently is broken for Postgres, MariaDB, and possibly others

## Steps To Reproduce
Include the following change in any liquibase migration on Postgres
`<alterSequence cycle="true" schemaName="public" sequenceName="sequence_that_exists"/>`

## Actual Behavior
SQL Generated:
`ALTER SEQUENCE public.sequence_that_exists NOCYCLE;`
This is not valid in [postgres](https://www.postgresql.org/docs/10/sql-altersequence.html), [mariadb/mysql](https://mariadb.com/kb/en/alter-sequence/), or [MSSQL](https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-sequence-transact-sql?view=sql-server-ver15)

## Expected/Desired Behavior
Expected SQL generated:
`ALTER SEQUENCE public.sequence_that_exists NO CYCLE;`

## Additional Context
Possibly something like [this](https://github.com/liquibase/liquibase/commit/8961e10620aef247ee92134b82f9121700cee66b) needs to happen. I don't have a java environment set up, so I didn't build/test this. This is just my shot in the dark. If this is the right spot, it might be good to cross-check multiple supported DBs for their `NOCYCLE` syntax.

Thanks for looking!

## Test Requirements (Liquibase Internal QA)
**Manual Tests**

One Oracle instance and one Postgres instance is necessary for these tests. The generated SQL for an alter sequence with `cycle="true”` differs slightly between the two platforms; valid SQL for Oracle requires the attribute `NOCYCLE` whereas Postgres requires `NO CYCLE`. Validation is to ensure the reported bug with Postgres is fixed and also to verify the fix does not introduce a regression for Oracle alter sequence statements. Use the attached changelog for these tests.

Note that all structured changelogs (XML, JSON, YML) go through the same code to generate SQL. These tests are changelog format agnostic. An XML changelog to use is attached.

_Verify a update~~sql generates platform~~specific SQL to alter a sequence._

`liquibase update-sql --changelog-file LB2111-changelog.xml`

* Expected SQL for Postgres:
  * `ALTER SEQUENCE public.test_sequence1 CYCLE;`
  * `ALTER SEQUENCE public.test_sequence2 NO CYCLE;`
* Expected SQL for Oracle: 
  * `ALTER SEQUENCE TEST_SEQUENCE1 CYCLE;`
  * `ALTER SEQUENCE TEST_SEQUENCE2 NOCYCLE;`

_Verify successful update to alter a sequence cycle value._

`liquibase update --changelog-file LB2111-changelog.xml`

* Postgres
* Oracle

**Automated Tests**

* No new functional automated tests required for this ticket.
* A new test harness test is required for this ticket.
* Add an XML changelog with two changesets:
  * createSequence with cycle=true
  * alterSequence with cycle=false
  * Note: There is already an alterSequence changelog; name yours alterSequenceCycleProperty.xml
  * liquibase-test-harness/src/main/resources/liquibase/harness/change/changelogs
* Add the JSON snapshot for test validation:
  * src/main/resources/liquibase/harness/change/expectedSnapshot/
* Add the generated SQL for test validation:
  * src/main/resources/liquibase/harness/change/expectedSql/oracle
  * src/main/resources/liquibase/harness/change/expectedSql/postgres
  * src/main/resources/liquibase/harness/change/expectedSql/mssql
  * src/main/resources/liquibase/harness/change/expectedSql/mariadb
    * Note : MariaDB began supporting sequences in 10.3. The test-harness runs versions of MariaDB starting with 10.2. You may need to create a 10.2 MariaDB subdirectory and add a file with the same name as the other expectedSql files (alterSequenceCycleProperty.sql). The body of the MariaDB altersequenceCycleProperty.sql should be the same as:
[https://github.com/liquibase/liquibase-test-harness/blob/main/src/main/resources/liquibase/harness/change/expectedSql/mariadb/alterSequence.sql](https://github.com/liquibase/liquibase-test-harness/blob/main/src/main/resources/liquibase/harness/change/expectedSql/mariadb/alterSequence.sql|smart-link) 
* Validate that Liquibase correctly generates NOCYCLE/NO CYCLE SQL for each of these platforms.

┆Issue is synchronized with this [Jira Story](https://datical.atlassian.net/browse/LB-2111) by [Unito](https://www.unito.io)

**Comment by molivasdat** [comment]
Hi @JayWelborn Thanks for bringing this issue up.  We will add this to the list of issues to process.  It is broken but want to make sure we fix it correctly.
To make sure that we have the correct scenario though. 

Above you have in your recreation scenario **cycle=true** means that when you hit the maximum or min number in the sequence whether you go back to the min/max value or not depending if you are ascending/descending respectively.  

**true** means that you can cycle back through the sequence. Which by definition = **CYCLE** , which I believe is the default if you were to leave it off the alter statement.

**cycle=false** means you cannot cycle back through to min/max value, which means **NO CYCLE**
So to summarize
`
<alterSequence cycle="true" schemaName="public" sequenceName="sequence_that_exists"/>
`
should produce 
`
ALTER SEQUENCE public.sequence_that_exists CYCLE;
`
and 
`
<alterSequence cycle="false" schemaName="public" sequenceName="sequence_that_exists"/>
`
should produce 

`
ALTER SEQUENCE public.sequence_that_exists NO CYCLE;
`
Lastly since it's an optional field
`
<alterSequence schemaName="public" sequenceName="sequence_that_exists"/>
`
should produce
`
ALTER SEQUENCE public.sequence_that_exists
`

**Comment by sapanparikh18** [comment]
Hi @molivasdat 
Can you please review https://github.com/liquibase/liquibase/pull/2117? Feedback is welcome.
